### ISSUE TITLE
MyLocationView doesn't lie flat when the map is tilted

### ISSUE THREAD

**Issue by Kahtaf** [mapbox/mapbox-gl-native#5702]
**Platform: Android**
**Mapbox SDK version: 4.1.0**

### Steps to trigger behavior

Set a custom drawable for myLocationView using:

`map.getMyLocationViewSettings().setForegroundDrawable(userDot, userDot);`

Where `userDot` is a custom my location drawable. Then, set a new camera position that has a **tilt** using:

```
CameraUpdate cameraUpdate = CameraUpdateFactory.newCameraPosition(new CameraPosition.Builder().target(new LatLng(location.getLatitude(), location.getLongitude())).tilt(50f).build());

map.moveCamera(cameraUpdate);
```

The userDot drawable is not flat (ie: tilted in relation to the map), and looks out of place. Same behaviour if camera update set before myLocationView drawable is set.

### Expected behavior
When the map has a tilt, I would expect the myLocationView drawable would lie flat against the ground, which it currently does not. (image 1). 

### Actual behavior
Initially, it looks like how it would behave if the map didn't have a tilt. However, if I tilt the map with my fingers using tilt gestures, the drawable immediately flattens (image 2). This is how I expect it to look when I first load the map and set the tilt.

![screenshot_20160715-140212](https://cloud.githubusercontent.com/assets/5796181/16884163/e341b9ae-4a95-11e6-97b8-73db04323785.png)
![screenshot_20160715-140221](https://cloud.githubusercontent.com/assets/5796181/16884162/e318fde8-4a95-11e6-83fc-25d6b47e9491.png)

**Comment by cammace** [comment]
Looks like this bug might have been fixed recently, can you try using the `4.2.0-SNAPSHOT` and see if the issue still occurs. Here's how you switch SDKs in the apps `build.grade` file:

```groovy
repositories {
    mavenCentral()
    maven { url "http://oss.sonatype.org/content/repositories/snapshots/" }
}

dependencies {
    compile ('com.mapbox.mapboxsdk:mapbox-android-sdk:4.2.0-SNAPSHOT@aar'){
        transitive=true
    }
}
```

The fix seems to work good when you move the camera but easing or animating the camera while the user location's enabled could still be improved shown in this GIF:

![ezgif com-crop](https://cloud.githubusercontent.com/assets/5652865/16904043/875960ba-4c5c-11e6-9baf-69dc71949b5a.gif)

@tobrun anyway we could animate the location marker tilting as well instead of redrawing at the end of the animation?

**Comment by Kahtaf** [comment]
`4.2.0-SNAPSHOT` works, thanks!
